api/src/movielist/movielist.service.ts

```diff
@@ -19,15 +19,29 @@ export class MovielistService {
 
   async cloneList(listId: string, userId: string, name?: string) {
     const originalList = await this.listService.getList(listId);
-    const { movies } = await this.movieService.getMovies({ listId });
-
-    const clonedList = await this.createListWithMovies(
-      {
-        name: name || originalList.name,
-        movie: movies,
-      },
+    const clonedList = await this.listService.createList(
+      name || originalList.name,
       userId,
     );
+    let { count } = await this.movieService.getMovies({ listId });
```

Comment:
Since we have the number of elements per page and the number of pages, I think we could calculate the total amount of elements and get them all in 2 requests, one for (num of pages - 1), and another one for the last page.
What do you think?